eslint.config.mjs

```diff
@@ -0,0 +1,130 @@
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+
+import { fixupPluginRules } from '@eslint/compat'
+import { FlatCompat } from '@eslint/eslintrc'
+import js from '@eslint/js'
+import tsParser from '@typescript-eslint/parser'
+import _import from 'eslint-plugin-import'
+import jsxA11Y from 'eslint-plugin-jsx-a11y'
+import observation from 'eslint-plugin-observation'
+import prettier from 'eslint-plugin-prettier'
+import react from 'eslint-plugin-react'
+import reactNative from 'eslint-plugin-react-native'
+import globals from 'globals'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+})
+
+export default [
+  ...compat.extends('plugin:@typescript-eslint/recommended'),
```

Comment:
Eerst extenden we van `@react-native` nu deze. Waarom?

Reply:
Voor zover ik het begrijp: extends bestaat niet meer sinds eslint 8.0.0. In plaats daarvan bestaat de configuratie uit een array van configuratie-objecten. Er is een utility `FileCompat` om hieraan eslint-configuraties toe te voegen die nog geen flat eslint-configuration hebben. De helper `fixupPluginRules` voor plugins doet iets dergelijks, is meer low level, en wordt intern door `FileCompat` gebruikt.

De configuratie in deze PR werkt voor de eslint packages die wij gebruiken. Als alle eslint packages support voor eslint 8.0.0 hebben gekregen, dan hebben we `FlatCompat` en `fixupPluginRules` niet meer nodig.